libbeat/common/kubernetes/k8skeystore/kubernetes_keystore_test.go

```diff
@@ -35,32 +41,169 @@ func TestGetKeystore(t *testing.T) {
 	assert.NotEqual(t, k2, k3)
 }
 
-// TODO: upgrade client dependency and use fake client to test retrieve
-//func TestGetKeystoreAndRetrieve(t *testing.T) {
-//	client := k8sfake.NewSimpleClientset()
-//	ns := "test_namespace"
-//	pass := "testing_passpass"
-//	secret := &v1.Secret{
-//		TypeMeta: metav1.TypeMeta{
-//			Kind:       "Secret",
-//			APIVersion: "apps/v1beta1",
-//		},
-//		ObjectMeta: metav1.ObjectMeta{
-//			Name:      "testing_secret",
-//			Namespace: ns,
-//		},
-//		Data: map[string][]byte{
-//			"secret_value": []byte(pass),
-//		},
-//	}
-//	client.CoreV1().Secrets(ns).Create(context.TODO(), secret, metav1.CreateOptions{})
-//
-//	kRegistry := NewKubernetesKeystoresRegistry(nil, nil)
-//	k1 := kRegistry.GetKeystore(bus.Event{"kubernetes": common.MapStr{"namespace": ns}})
-//	key := "kubernetes.test_namespace.testing_secret.secret_value"
-//	secretVal, err := k1.Retrieve(key)
-//	if err != nil {
-//		t.Fatalf("could not retrive k8s secret", err)
-//	}
-//	assert.Equal(t, pass, secretVal)
-//}
+func TestGetKeystoreAndRetrieve(t *testing.T) {
+	client := k8sfake.NewSimpleClientset()
+	ns := "test_namespace"
+	pass := "testing_passpass"
+	secret := &v1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "apps/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testing_secret",
+			Namespace: ns,
+		},
+		Data: map[string][]byte{
+			"secret_value": []byte(pass),
+		},
+	}
+	_, err := client.CoreV1().Secrets(ns).Create(context.TODO(), secret, metav1.CreateOptions{})
```

Comment:
Here and in other tests added in this PR it could be `context.Background()` :slightly_smiling_face: Because we are in tests, and initializing the top-level context.
```suggestion
	_, err := client.CoreV1().Secrets(ns).Create(context.Background(), secret, metav1.CreateOptions{})
```